code/modules/client/login/client_login.dm

```diff
@@ -114,6 +114,7 @@
 	if(first_run)
 		GLOB.clients_unsafe += src
 	GLOB.directory[ckey] = src
+	GLOB.connected_ckeys |= ckey
```

Comment:
Uuh, I am not sure if it's alright or added ckey would become null in the list while it doesn't disappear...??

Reply:
ckey is a string. Once it gets added it stays there forever. A string is an immutable object.

Reply:
ah, right it was ckey. I was mislooking it as client src.